art_tree.go

```diff
@@ -17,22 +18,44 @@ func NewArtTree() *ArtTree {
 	return &ArtTree{root: nil, size: 0}
 }
 
+// Finds the starting node for a prefix search and returns an array of all the objects under it
+func (t *ArtTree) PrefixSearch(key []byte) []interface{} {
```

Comment:
this version is slightly different... the 2 search helper functions were pretty similar, so i altered the existing one rather than adding another